in_toto/runlib.py

```diff
@@ -250,36 +249,18 @@ def execute_link(link_cmd_args, record_streams):
     if specified.
 
   <Returns>
-    - A dictionary containg standard output and standard error of the
+    - A dictionary containing standard output and standard error of the
       executed command, called by-products.
       Note: If record_streams is False, the dict values are empty strings.
     - The return value of the executed command.
   """
-  # TODO: The first approach only redirects the stdout/stderr to a tempfile
-  # but we actually want to duplicate it, ideas
-  #  - Using a pipe won't work because processes like vi will complain
-  #  - Wrapping stdout/sterr in Python does not work because the suprocess
-  #    will only take the fd and then uses it natively
-  #  - Reading from /dev/stdout|stderr, /dev/tty is *NIX specific
-
-  # Until we come up with a proper solution we use a flag and let the user
-  # decide if s/he wants to see or store stdout/stderr
-  # btw: we ignore them in the layout anyway
-
+  # TODO: Properly duplicate standard streams (issue #11)
   if record_streams:
-    # We should consider using SpooledTemporaryFile if we expect very large
-    # outputs in the future
-    stdout_file = tempfile.TemporaryFile()
-    stderr_file = tempfile.TemporaryFile()
-
-    return_value = subprocess.call(link_cmd_args,
-        stdout=stdout_file, stderr=stderr_file)
-
-    stdout_file.seek(0)
-    stderr_file.seek(0)
+    process = subprocess.Popen(link_cmd_args, stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE, universal_newlines=True)
 
-    stdout_str = stdout_file.read()
-    stderr_str = stderr_file.read()
+    stdout_str, stderr_str = process.communicate()
```

Comment:
I thought we had opted to not do this due to performance concerns/buffering. Is this still the case?

Reply:
I timed the function call for `ls -Rl /` and it executed without error in about 4 minutes, returning a stdout string of about 170MB.

```python
>>> import timeit
>>> timeit.Timer("in_toto.runlib.execute_link(['ls', '-Rl', '/'], True)", setup="import in_toto.runlib").timeit(1)
247.98906087875366
```

The timed `ls -Rl` in bash takes longer probably due to writing the ouput to the terminal.
```shell
$ time ls -Rl /
real  4m57.299s
user  0m51.453s
sys 2m20.026s
```

I think it's fine for now. Also it is only a temporary solution, if we decide to adapt https://github.com/in-toto/in-toto/pull/160